server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1024,12 +1024,11 @@ public long getRestoreThrottleTimeInNanos() {
         return restoreRateLimitingTimeInNanos.count();
     }
 
-    protected void assertSnapshotOrGenericThread() {
-        // NORELEASE
-        /*
-        assert Thread.currentThread().getName().contains(ThreadPool.Names.SNAPSHOT)
-            || Thread.currentThread().getName().contains(ThreadPool.Names.GENERIC) :
-            "Expected current thread [" + Thread.currentThread() + "] to be the snapshot or generic thread.";*/
+    protected void assertUsingPermittedThreadPool() {
+        assert Thread.currentThread().getName().contains('[' + ThreadPool.Names.SNAPSHOT + ']')
+            || Thread.currentThread().getName().contains('[' + ThreadPool.Names.GENERIC + ']')
+            || Thread.currentThread().getName().contains('[' + ThreadPool.Names.SEARCH + ']')
```

Comment:
Hmm I don't think this works, we should be seeing this fail due to access on the `search_throttled` threadpool and also (TBC) the can-match phase runs elsewhere too.